newIDE/app/src/VariablesList/VariablesEditorDialog.js

```diff
@@ -36,24 +53,79 @@ const VariablesEditorDialog = ({
   title,
   emptyPlaceholderTitle,
   emptyPlaceholderDescription,
+  project,
   variablesContainer,
   inheritedVariablesContainer,
   hotReloadPreviewButtonProps,
   onComputeAllVariableNames,
   helpPagePath,
+  preventRefactoringToDeleteInstructions,
   id,
 }: Props) => {
+  const { showConfirmation } = useAlertDialog();
   const {
     onCancelChanges,
     notifyOfChange,
+    getOriginalContentSerializedElement,
   } = useSerializableObjectCancelableEditor({
     serializableObject: variablesContainer,
     onCancel,
+    resetThenClearPersistentUuid: true,
   });
   const { DismissableTutorialMessage } = useDismissableTutorialMessage(
     'intro-variables'
   );
 
+  const onRefactorAndApply = React.useCallback(
+    async () => {
+      if (inheritedVariablesContainer) {
+        // No refactoring to do - this is a variable container of an instance
+        // (or something else that overrides variables from another container),
+        // which does not have an impact on the rest of the project.
+      } else {
+        const changeset = gd.WholeProjectRefactorer.computeChangesetForVariablesContainer(
+          project,
+          getOriginalContentSerializedElement(),
+          variablesContainer
+        );
+        if (preventRefactoringToDeleteInstructions) {
+          // Clear the removed variables from the changeset, so they do not trigger
+          // deletion of actions/conditions or events using them.
+          changeset.clearRemovedVariables();
+        }
+        if (changeset.hasRemovedVariables()) {
+          const shouldRemoveVariables = await showConfirmation({
+            title: t`Remove actions and conditions?`,
+            message: t`You've removed some variables. Do you want to also remove all the actions and conditions in events using them?`,
+            confirmButtonLabel: t`Don't remove anything else`,
```

Comment:
same remark here

Reply:
As the other, I'll remove this dialog (and the removal of actions/conditions) entirely.